### ISSUE TITLE
Could you add declarativewidgets as part of the playground?

### ISSUE THREAD

**Issue by lbustelo** [bollwyvl/dangerous-playgrounds#1]
@bollwyvl Would love to see https://github.com/jupyter-incubator/declarativewidgets represented in your playground. We actually did a couple of examples where we combined Bokeh and declarativewidgets.
- http://jupyter.cloudet.xyz/notebooks/declarativewidgets/examples/BokehUrthBasic.ipynb
- http://jupyter.cloudet.xyz/notebooks/declarativewidgets/examples/ArxivBokeh.ipynb

Reach out with any questions, or let me know if you are open to a PR.

**Comment by bollwyvl** [comment]
Sorry I just saw this! I didn't know anyone had looked at it!

In a sense, a playground is kind of the TodoMVC of reactive programming. I'd certainly be open to seeing other approaches to reaching the same goal. 

The unstated idea on this little project was to get to as simple an example as possible of a non-trivial widget implementation that could rapidly replace a capability found somewhere on the web. It's a pretty long page, for sure, but I was able to author the whole thing in the notebook in a few hours, which was also part of the experiment. Indeed, being in the notebook, while making debugging worse, makes deployment better, as the speed of prototyping can't really be beaten... especially when you stir in something like this:
https://gist.github.com/bollwyvl/10440652

Thus, I have been put off by the polymer stuff in general because i really can't stand authoring HTML source directly... hangover from the cold fusion days. Have you guys explored any other compile-to-html languages to stand in for %%html? jade is a personal favorite of mine. Further, if you are actually expecting users.

Actually, having a playground that wrapped up the [es2015 example](https://codelabs.developers.google.com/codelabs/polymer-es2015/#3), along with jade/haml would actually make me think more about getting interested!

**Comment by lbustelo** [comment]
Take a look at this example when you get a chance. http://jupyter.cloudet.xyz/notebooks/declarativewidgets/demos/Todo_List_Tutorial.ipynb

Your comments about building an app on the notebook... Man, you are preaching to the choir. I'm totally there and that is why I would like you to take a closer look at the declarativewidgets project.

I'm prototyping ideas for projects completely on notebooks, and declarativewidgets helps a ton. 

I hear your concern about %%html and I'm open to suggestions to provide alternative authoring techniques.

**Comment by bollwyvl** [comment]
> Take a look at this example when you get a chance. http://jupyter.cloudet.xyz/notebooks/declarativewidgets/demos/Todo_List_Tutorial.ipynb

Yeah, looks good. With respect to the `channel` stuff, I am personally of a mind that I would prefer to have more, richer stateful widget models than deeper integration into kernel stuff. The first important piece of that is the the dataframe/series traitlet work, but there are other things going on around.

The cadillac, in my opinion, would be something derived from this:
https://developers.google.com/web/updates/2015/03/creating-semantic-sites-with-web-components-and-jsonld?hl=en

> I hear your concern about %%html and I'm open to suggestions to provide alternative authoring techniques. 

Well, I think Jade is definitely worth exploring: I whacked [this magic](https://github.com/bollwyvl/jademagic) together in a half hour from a [previous thing](https://github.com/bollwyvl/yamlmagic). With it, one could write, instead:

``` jade
link(rel='import', href='urth_components/paper-card/paper-card.html', is='urth-core-import', package='PolymerElements/paper-card')
link(rel='import', href='urth_components/paper-item/paper-item.html', is='urth-core-import', package='PolymerElements/paper-item')
link(rel='import', href='urth_components/paper-input/paper-input.html', is='urth-core-import', package='PolymerElements/paper-input')
link(rel='import', href='urth_components/paper-icon-button/paper-icon-button.html', is='urth-core-import', package='PolymerElements/paper-icon-button')
link(rel='import', href='urth_components/paper-button/paper-button.html', is='urth-core-import', package='PolymerElements/paper-button')
link(rel='import', href='urth_components/iron-icons/iron-icons.html', is='urth-core-import', package='PolymerElements/iron-icons')
link(rel='import', href='urth_components/iron-collapse/iron-collapse.html', is='urth-core-import', package='PolymerElements/iron-collapse')
style(is='custom-style').
  paper-card {
  width: 100%;
  }
  .list {
  @apply(--layout-vertical);
  }
template(is='urth-core-bind')
  paper-card(heading='Todo List', style='width: 100%')
    .card-content
      .list(role='listbox')
        template(is='dom-repeat', items='{{todo}}')
          paper-item
            urth-core-function(ref='removeTodo', arg-idx='[[index]]', onclick='this.invoke()', delay='250')
              paper-icon-button(icon='clear') X
            span {{item.0}}
            template(is='dom-if', if='{{item.1}}')
              span(style='color: gray')  ({{item.1}})
        div
        .card-actions
          paper-button(onclick="newItemForm.toggle(); titleIn2.value=''; dueDateIn2.value=''") New Item
          iron-collapse#newItemForm
            .content
              paper-input#titleIn2(label='Title', value='{{title}}')
              paper-input#dueDateIn2(label='Due Date', value='{{dueDate}}')
              div(style='text-align: right; margin: 10px 20px')
                urth-core-function(ref='addTodo', arg-title='{{title}}', arg-date='{{dueDate}}', onclick='this.invoke();', delay='250')
                  paper-button(raised='', onclick='newItemForm.toggle()')
                    iron-icon(icon='add')
                    | Add Item
```

With a touch more work, one could have even shorter syntax for certain key activites.

**Comment by lbustelo** [comment]
I like the idea... I wonder how something like this can be done without kernel specific code (i.e. the magic). I'm assuming that all kernels should at least have an `%%html` magic. What we really need is to have client side magics that can bypass the kernel.

**Comment by bollwyvl** [comment]
Yeah, very true.

I think you _could_ do it today, by means of creating an empty (hidden)
cell with a js mimebundle in an output. Then make your own input cell,
which would just pass the content directly to the output. But that's very
notebook specific.

I think in JupyterLab, with native npm, something more robust will be
feasible: you'd still want to deliver the dependencies to the browser a la
webpack/browserify/require. Since these things would, by definition, not
interact with the kernel, they could be much further out on the playground
spectrum.
